test/rekt/resources/eventprober/prober.go

```diff
@@ -0,0 +1,271 @@
+	"knative.dev/reconciler-test/pkg/manifest"
+)
+
+func New(targetGVR schema.GroupVersionResource, targetName, targetURI string) *EventProber {
+	return &EventProber{
+		target: target{
+			gvr:  targetGVR,
+			name: targetName,
+			uri:  targetURI,
+		},
+		shortNameToName: make(map[string]string),
+	}
+}
+
+type EventProber struct {
+	target          target
+	shortNameToName map[string]string
+	ids             []string
+	senderOptions   []eventshub.EventsHubOption
+}
+
+type target struct {
+	// Need [GVR + Name] OR [URI]
+	gvr  schema.GroupVersionResource
+	name string
+	uri  string
+}
+
+type EventInfoCombined struct {
+	Sent     EventInfo
+	Response EventInfo
+}
+
+func (p *EventProber) RxInstall(prefix string) feature.StepFn {
+	name := feature.MakeRandomK8sName(prefix)
+	p.shortNameToName[prefix] = name
+	return eventshub.Install(name, eventshub.StartReceiver)
+}
+
+func (p *EventProber) TxDone(prefix string) feature.StepFn {
+	//	name := p.shortNameToName[prefix]
+	return func(ctx context.Context, t feature.T) {
+		interval, timeout := environment.PollTimingsFromContext(ctx)
+		err := wait.PollImmediate(interval, timeout, func() (bool, error) {
+			events := p.SentBy(ctx, prefix)
+			if len(events) == len(p.ids) {
+				return true, nil
+			}
+			return false, nil
+		})
+		if err != nil {
+			t.Failed()
+		}
+	}
+}
+
+func (p *EventProber) TxInstall(prefix string) feature.StepFn {
+	name := feature.MakeRandomK8sName(prefix)
+	p.shortNameToName[prefix] = name
+	return func(ctx context.Context, t feature.T) {
+		var opts []eventshub.EventsHubOption
+		if len(p.target.uri) > 0 {
+			opts = append(p.senderOptions, eventshub.StartSenderURL(p.target.uri))
+		} else if !p.target.gvr.Empty() {
+			opts = append(p.senderOptions, eventshub.StartSenderToResource(p.target.gvr, p.target.name))
+		} else {
+			t.Fatal("no target is configured for event loop")
+		}
+		// Install into the env.
+		eventshub.Install(name, opts...)(ctx, t)
+	}
+}
+
+// CorrelateSent takes in an array of mixed Sent / Response events (matched with sentEventMatcher for example)
+// and correlates them based on the sequence into a pair.
+func CorrelateSent(origin string, in []EventInfo) []EventInfoCombined {
+	var out []EventInfoCombined
+	// not too many events, this will suffice...
+	for i, e := range in {
+		if e.Origin == origin && e.Kind == EventSent {
+			looking := e.Sequence
+			for j := i + 1; j <= len(in)-1; j++ {
+				if in[j].Kind == EventResponse && in[j].Sequence == looking {
+					out = append(out, EventInfoCombined{Sent: e, Response: in[j]})
+				}
+			}
+		}
+	}
+	return out
+}
+
+func (p *EventProber) SentBy(ctx context.Context, prefix string) []EventInfoCombined {
+	name := p.shortNameToName[prefix]
+	store := eventshub.StoreFromContext(ctx, name)
+
+	for _, c := range store.Collected() {
+		fmt.Printf("%#v\n", c)
+	}
+
+	return CorrelateSent(name, store.Collected())
+}
+
+func (p *EventProber) ReceivedBy(ctx context.Context, prefix string) []EventInfo {
```

Comment:
case in point (I think) about Rx vs. ReceivedBy. I think using consistent method names would be clearer.